### ISSUE TITLE
[bug] Level 8.0.0 throws error "db.createValueStream is not a function"

### ISSUE THREAD

**Issue by seriousme** [mcollina/aedes-persistence-level#27]
**System Information**
- Aedes: 0.46.3
- NodeJS: 16.14.2
- OS: Debian GNU/Linux 11 (bullseye)
- Arch: x86_64

**Describe the bug**
After upgrading to  `level` to 8.0.0 `aedes-persistence-level` throws: `TypeError: db.createValueStream is not a function`

**To Reproduce**
```js
import { Level } from "level";
import aedesPersistencelevel from "aedes-persistence-level";
const db = aedesPersistencelevel(new Level("./data"));
```
**Expected behavior**
No error :-)

**Additional context**
Level 8.0.0 [UPGRADING.md](https://github.com/Level/level/blob/HEAD/UPGRADING.md) mentions this as a breaking change for 8.0.0.:

>Streams have moved
>Node.js readable streams must now be created with a new standalone module called [level-read-stream](https://github.com/Level/read->stream) rather than database methods like db.createReadStream(). For browsers you might prefer [level-web-stream](https://github.com/Level/web-stream) which does not require bundling the [buffer](https://github.com/feross/buffer) or [readable-stream](https://github.com/nodejs/readable-stream)[](https://github.com/Level/level/blob/HEAD/UPGRADING.md#state-checks-for-safety) shims. Both level-read-stream and level-web-stream can be used in Node.js and browsers. The former is significantly faster (also compared to level@7, thanks to a new nextv() method on iterators). The latter is a step towards a standard library for JavaScript across Node.js, Deno and browsers.
>
>To offer an alternative to db.createKeyStream() and db.createValueStream(), two new types of iterators have been added: db.keys() and db.values().
```

**Comment by robertsLando** [comment]
PR?

**Comment by seriousme** [comment]
Happy to, but how do you want to move forward ?
E.g. Move to aedes-persistence-level version 8 that is only compatible with Level version 8, or some auto detection logic ?
Also do you want to modernize to iterators as well or do you want to stay on level-read-stream/level-web-stream ?

https://github.com/moscajs/aedes-persistence-level/blob/8600507ebb33e8088b51a95e49b5d1cd1d20b9b2/persistence.js#L29
Could still return a stream if required but based on iterators ;-)

Quick scan seems to show that  Aedes does not seem to use `createValueStream` therefore we could also switch all the internals from streams to iterators.
And if it will be a semver major, do you also want to move to classes like `level` did ?

So the question is: how far do you want to go in refurbishing this module ;-)

Kind regards,
Hans

**Comment by robertsLando** [comment]
I'm +1 with all your suggestions. 

Check if it's possible to support also older level versions, if not just make it compatible with level 8+, we will release a major of this package in that case.

This package didn't get so much attention in last year so feel free to do all the updates you want (just keep it compatible with aedes-persistence and aedes)

**Comment by seriousme** [comment]
Ok, any other wishes you'd like me to take along ? 
Minimum Node.js version 14  ? (12 is EOL by the end of this month ;-))

Cheers,
Hans

**Comment by robertsLando** [comment]
> Minimum Node.js version 14 ? (12 is EOL by the end of this month ;-))

Yep it's ok for me 👍🏼  Remember to update ci workflows

**Comment by seriousme** [comment]
>Remember to update ci workflows

Ok, will look into those as well :-)

**Comment by seriousme** [comment]
Small update: 

I got a [working version](https://github.com/seriousme/aedes-persistence-level/tree/level8) that supports `Level` versions: 6.0.1, 7.0.1, 8.0.0
(6.0.1 being the version used in CI for the current version of `aedes-persistence-level`)

In the mean time I learned quite a lot including:
- level.createReadStream in 6.0.1 and 7.0.1 does not create an iterable stream (its absent in 8.0.0)
- level.iterator in 6.0.1 creates an iterator that is not standard
- 'utf8' encoding in 6.0.1 and 7.0.1 did not touch buffers, in 8.0.0 it does so now all db.get and db.put calls have correct encoding

Currently I'm working to move the streams to the edge of the interface using generators and iterables.
This should also reduce the number of dependencies. 

Cheers,
Hans

**Comment by robertsLando** [comment]
Open a PR so I can follow the changes

**Comment by seriousme** [comment]
Fixed by #28
